sim/src/main/cc/midasexamples/TerminationModule.h

```diff
@@ -0,0 +1,63 @@
+//See LICENSE for license details.
+
+#include "simif_peek_poke.h"
+#include "bridges/termination.h"
+#include <iostream>
+#include <string.h>
+
+class TerminationModule_t: public simif_peek_poke_t
+{
+public:
+  termination_t * terminator;
+  TerminationModule_t(int argc, char** argv) {
+    TERMINATIONBRIDGEMODULE_0_substruct_create;
+    std::vector<std::string> args(argv + 1, argv + argc);
+    terminator = new termination_t(
+      this, args,
+      TERMINATIONBRIDGEMODULE_0_substruct,
+      TERMINATIONBRIDGEMODULE_0_message_count,
+      TERMINATIONBRIDGEMODULE_0_message_type,
+      TERMINATIONBRIDGEMODULE_0_message);
+  };
+  void run() {
+    poke(reset, 1);
+    //poke(io_globalReset, 1);
+    int terminations = 0;
+    int lv_validinCycle = 0;
+    int validinCycle = 0;
+    int lv_msginCycle = 0;
+    int msginCycle = 0;
+    int msgid = 0;
+    char failure_list[3][10] = {"failure 1",
+                                "failure 2",
+                                "failure 3"};
+    bool prematureExit = true;
+    validinCycle = rand_next(100);
+    msginCycle = rand_next(100);
+    int errCode = rand_next(8);
+    lv_validinCycle = lv_validinCycle + validinCycle;
+    lv_msginCycle = lv_msginCycle + msginCycle;
+    poke(io_validInCycle, lv_validinCycle);
+    poke(io_msgInCycle, lv_msginCycle);
+    poke(io_doneErrCode, errCode);
+    //terminator->resume();
+    step(1);
+    //poke(io_globalReset, 0);
+    poke(reset, 0);
+    if(errCode %2 !=0) {
+      msgid = 0;
+    } else if(errCode %4 != 0) {
+      msgid = 1;
+    } else {
+      msgid = 2;
+    }
+    step(lv_validinCycle+2, false);
+    while (!terminator->terminate()) {
+      terminator->tick();
+    }
+    int str_match = strcmp(terminator->exit_message(),failure_list[msgid]);
+    expect(terminator->cycleCount()==(lv_validinCycle+2), "Code Exits at precise time");
+    expect(terminator->exit_code()==(msgid), "Error Code Verified");
```

Comment:
The exit code should be either 0 or 1, i.e, it should be the result of `isErr[msg_id]`

If one day we want to better capture _why_ something failed we'll have to log it to a file with some more info about the bridge.